src/rqt_service_caller/service_caller.py

```diff
@@ -40,7 +40,8 @@ def __init__(self, context):
         super(ServiceCaller, self).__init__(context)
         self.setObjectName('ServiceCaller')
 
-        self._widget = ServiceCallerWidget()
+        assert hasattr(context, 'node'), 'Context does not have a node.'
```

Comment:
By declaring this plugin to be an rqt_gui plugin in the package.xml, plugin_context here will be the ros2 one.

Reply:
Ideally yes. I had run into a corner case where this failed silently so I added an assert to make it easier to track down in case it happens again. I don't see a reason to remove it.